src/GafferScene/SceneReader.cpp

```diff
@@ -455,7 +479,8 @@ void SceneReader::plugSet( Gaffer::Plug *plug )
 
 ConstSceneInterfacePtr SceneReader::scene( const ScenePath &path ) const
 {
-	std::string fileName = fileNamePlug()->getValue();
+	std::string fileName = fileNameValue( Context::current() );
```

Comment:
These `Context::current()` lookups aren't quite as free as we'd like - could you modify the signature for `scene()` so that the context we already have is passed to it?